client/src/RestFileUploader.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Libero\SharedServicesExperiment\Client;
+
+use InvalidArgumentException;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Exception\GuzzleException;
+use RuntimeException;
+
+use function GuzzleHttp\Psr7\stream_for;
+
+class RestFileUploader implements FileUploader
+{
+    /**
+     * Http Client
+     *
+     * @var ClientInterface
+     */
+    private $client;
+
+    /**
+     * Construct the client instance
+     *
+     * @param ClientInterface $client
+     */
+    public function __construct(ClientInterface $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Upload a file
+     *
+     * @param string $sourcePath
+     * @param string $uploadPath
+     * @return FileUploadRecord
+     * @throws InvalidArgumentException
+     * @throws FileUploaderException
+     */
+    public function uploadFile(string $sourcePath, string $uploadPath): FileUploadRecord
+    {
+        if (! file_exists($sourcePath)) {
+            throw new InvalidArgumentException('File not found: ' . $sourcePath);
+        }
+
+        $finfo         = finfo_open(FILEINFO_MIME_TYPE);
```

Comment:
a library like `symfony/filesystem` would probably be more realistic, albeit with more overhead?

Reply:
If needed, can be added further down the line. Not sure what the advantage would be in this case

Reply:
Possibly, but not sure whats he advantage at this stage.